libmobilecoin/include/transaction.h

```diff
@@ -515,9 +537,202 @@ bool mc_memo_sender_with_payment_request_memo_get_payment_request_id(
 )
 MC_ATTRIBUTE_NONNULL(1, 2);
 
+/* ==== Gift Code Memo Builders ==== */
 
-/* ==== Decrypt Memo Payload ==== */
+/// # Preconditions
+///
+/// * `gift_code_funding_note` - must be a null-terminated C string containing
+/// up to 54 valid UTF-8 bytes. The actual note stored on chain is up to 53 null
+/// terminated UTF-8 bytes unless the note is exactly 53 utf-8 bytes long,
+/// in which case, no null bytes are stored. If the C string passed here is
+/// exactly 54 bytes, the last byte MUST be null and that byte will be
+/// removed prior to storage on chain.
```

Comment:
I had considered that the verbosity here was explaining too much - but I sort of think it's necessary to explain the subtle nuance so that people mobile sdk developers aren't making assumptions that could cause errors. It's kind of a hard to understand distinction, so I think it's worth making developers take a pause to understand it.

Reply:
@the-real-adammork is the maintainer of this - let's see what he think about this

Reply:
what happens if the string is longer than 54 bytes ? WIll it throw an error or will it only accept the first 53 bytes